src/helpers/string_helpers.sh

```diff
@@ -3,6 +3,18 @@
 # Helper functions for working with strings
 
 
+function contains_string {
+  local haystack=$1
+  # Note: cannot assign $2 to a variable here
+  #       because ShellCheck isn't smart enough to recognize the variable in the regex
```

Comment:
I'd rather ignore a shellcheck error instead of changing code since we think this is a missing feature / bug in shellcheck.